### PR TITLE
Add support for left padding

### PR DESCRIPTION
**Pull request by danieldk**
## Description

Left-padding is not strictly needed for transformer models, since the attention mask determines what tokens are attended to. However, left-padded sequences are slightly easier to reason about/debug in generation, since the input tokens immediately precede the generated tokens.

### Types of change

Feature

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by shadeMe — APPROVED** [review]

File: curated_transformers/tokenization/tokenizer.py [hunk 1]
```diff
@@ -13,25 +14,46 @@ class PiecesWithIds:
     ids: List[List[int]]
     pieces: List[List[str]]
 
-    @property
-    def attention_mask(self):
-        """CPU tensor with attention masks.
+    def attention_mask(self, pad_left: bool = False) -> Tensor:
```
Comment (shadeMe) [inline curated_transformers/tokenization/tokenizer.py:17]:
```suggestion
    def attention_mask(self, *, pad_left: bool = False) -> Tensor:
```
